eos/rare-b-decays/charm-loops.cc

```diff
@@ -1476,6 +1479,195 @@ namespace eos
         return result;
     }
 
+
+    /*
+     * Adaptater that exports the charn loops function as observables
+     */
+    template <>
+    struct Implementation<CharmLoopsAdaptater>
+    {
+        SpecifiedOption opt_contribution;
+
+        UsedParameter m_b;
+        UsedParameter m_c;
+        UsedParameter mu;
+
+        static const std::vector<OptionSpecification> options;
+
+        unsigned flag_0 = 0,
+                 flag_a = 0,
+                 flag_b = 0,
+                 flag_c = 0,
+                 flag_d = 0,
+                 flag_e = 0,
+                 flag_ctQc = 0,
+                 flag_ctQs = 0,
+                 flag_ctQb = 0;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            opt_contribution(o, options, "contribution"),
+            m_b(p["mass::b(MSbar)"], u),
+            m_c(p["mass::c"], u),
+            mu(p["sb::mu"], u)
+        {
+            if ("0" == opt_contribution.value())
+                flag_0 = 1;
+            if (("a" == opt_contribution.value()) || ("Qsb" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_a = 1;
+            if (("b" == opt_contribution.value()) || ("Qsb" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_b = 1;
+            if (("c" == opt_contribution.value()) || ("Qc" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_c = 1;
+            if (("d" == opt_contribution.value()) || ("Qc" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_d = 1;
+            if (("e" == opt_contribution.value()) || ("Qc" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_e = 1;
+            if (("ctQc" == opt_contribution.value()) || ("Qc" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_ctQc = 1;
+            if (("ctQs" == opt_contribution.value()) || ("Qsb" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_ctQs = 1;
+            if (("ctQb" == opt_contribution.value()) || ("Qsb" == opt_contribution.value()) || ("all" == opt_contribution.value()))
+                flag_ctQb = 1;
+        }
+
+        inline agv_2019a::CharmLoopsParameters clp(const complex<double> & s) const
+        {
+            return agv_2019a::CharmLoopsParameters(mu / m_b, s / m_b() / m_b(), (m_c * m_c) / (m_b * m_b), 1e-12);
+        }
+
+        complex<double> F17(const complex<double> & s) const
+        {
+            auto params = clp(s);
+            return (double)flag_a * agv_2019a::f17a(params)
```

Comment:
I would declare the flags as ``double`` from the get go, to avoid the casting here.